src/ripple/app/tx/impl/XChainBridge.cpp

```diff
@@ -0,0 +1,1675 @@
+
+    if (!accountSigners)
+    {
+        return {r, q, tecINTERNAL};  // LCOV_EXCL_LINE
+    }
+
+    for (auto const& as : *accountSigners)
+    {
+        r[as.account] = as.weight;
+    }
+
+    // add the master and regular keys. If they are already part of the signer's
+    // list, overwrite their weights.
+    if (masterKey)
+        r[*masterKey] = std::numeric_limits<std::uint16_t>::max();
+    if (regularKey)
+        r[*regularKey] = std::numeric_limits<std::uint16_t>::max();
+
+    return {std::move(r), q, tesSUCCESS};
+};
+}  // namespace
+//------------------------------------------------------------------------------
+
+NotTEC
+BridgeCreate::preflight(PreflightContext const& ctx)
+{
+    if (!ctx.rules.enabled(featureXChainBridge))
+        return temDISABLED;
+
+    if (auto const ret = preflight1(ctx); !isTesSuccess(ret))
+        return ret;  // LCOV_EXCL_LINE
+
+    if (ctx.tx.getFlags() & tfUniversalMask)
+        return temINVALID_FLAG;
+
+    auto const account = ctx.tx[sfAccount];
+    auto const reward = ctx.tx[sfSignatureReward];
+    auto const minAccountCreate = ctx.tx[~sfMinAccountCreateAmount];
+    auto const bridge = ctx.tx[sfXChainBridge];
+    if (bridge.lockingChainDoor() == bridge.issuingChainDoor())
+    {
+        return temEQUAL_DOOR_ACCOUNTS;
+    }
+
+    if (bridge.lockingChainDoor() != account &&
+        bridge.issuingChainDoor() != account)
+    {
+        return temSIDECHAIN_NONDOOR_OWNER;
+    }
+
+    if (isXRP(bridge.lockingChainIssue()) != isXRP(bridge.issuingChainIssue()))
+    {
+        // Because ious and xrp have different numeric ranges, both the src and
+        // dst issues must be both XRP or both IOU.
+        return temSIDECHAIN_BAD_ISSUES;
+    }
+
+    if (!isXRP(reward) || reward.signum() <= 0)
+    {
+        return temXCHAIN_BRIDGE_BAD_REWARD_AMOUNT;
+    }
+
+    if (minAccountCreate &&
+        ((!isXRP(*minAccountCreate) || minAccountCreate->signum() <= 0) ||
+         !isXRP(bridge.lockingChainIssue()) ||
+         !isXRP(bridge.issuingChainIssue())))
+    {
+        return temXCHAIN_BRIDGE_BAD_MIN_ACCOUNT_CREATE_AMOUNT;
+    }
+
+    if (isXRP(bridge.issuingChainIssue()))
+    {
+        // Issuing account must be the root account for XRP (which presumably
+        // owns all the XRP). This is done so the issuing account can't "run
+        // out" of wrapped tokens.
+        static auto const rootAccount = calcAccountID(
+            generateKeyPair(
+                KeyType::secp256k1, generateSeed("masterpassphrase"))
+                .first);
+        if (bridge.issuingChainDoor() != rootAccount)
+        {
+            return temSIDECHAIN_BAD_ISSUES;
+        }
+    }
+    else
+    {
+        // Issuing account must be the issuer for non-XRP. This is done so the
+        // issuing account can't "run out" of wrapped tokens.
+        if (bridge.issuingChainDoor() != bridge.issuingChainIssue().account)
+        {
+            return temSIDECHAIN_BAD_ISSUES;
+        }
```

Comment:
That makes sense. If the door is locking their own assets, in a sense they aren't locking anything. I added a new check, but I haven't pushed to the main branch yet - some unit tests need to be modified. But this will be resolved shortly.

Reply:
Fixed in e2592b695 [fold] Disallow locking a door account's issued assets

Reply:
I just verified this fix.  Thanks for taking care of that!